frontend/src/metabase/meta/types/Dataset.js

```diff
@@ -0,0 +1,22 @@
+/* @flow */
+
+export type Timestamp = string;
+
+export type ColumnName = string;
+
+// TODO: incomplete
+export type Column = {
+    name: ColumnName,
+    display_name: string,
+    base_type: string
+}
+
+export type Row = Array<string|number|Timestamp>
```

Comment:
Also, it would be super cool if we could export the Clojure schemas to Flow types automagically...